oauth2client/flask_util.py

```diff
@@ -473,24 +492,25 @@ def required(self, decorated_function=None, scopes=None,
         def curry_wrapper(wrapped_function):
             @wraps(wrapped_function)
             def required_wrapper(*args, **kwargs):
-
                 return_url = decorator_kwargs.pop('return_url', request.url)
 
-                # No credentials, redirect for new authorization.
-                if not self.has_credentials():
-                    auth_url = self.authorize_url(
-                        return_url,
-                        scopes=scopes,
-                        **decorator_kwargs)
-                    return redirect(auth_url)
+                requested_scopes = list(self.scopes) + (scopes or [])
```

Comment:
Use of "and" and "or" for non-boolean values is now considered low-class. Although it's longer, please go with `[] if scopes is None else scopes` instead of `scopes or []`.